word_language_model/generate.py

```diff
@@ -49,19 +49,41 @@
 
 corpus = data.Corpus(args.data)
 ntokens = len(corpus.dictionary)
-hidden = model.init_hidden(1)
-input = torch.randint(ntokens, (1, 1), dtype=torch.long).to(device)
-
-with open(args.outf, 'w') as outf:
-    with torch.no_grad():  # no tracking history
-        for i in range(args.words):
-            output, hidden = model(input, hidden)
-            word_weights = output.squeeze().div(args.temperature).exp().cpu()
-            word_idx = torch.multinomial(word_weights, 1)[0]
-            input.fill_(word_idx)
-            word = corpus.dictionary.idx2word[word_idx]
-
-            outf.write(word + ('\n' if i % 20 == 19 else ' '))
-
-            if i % args.log_interval == 0:
-                print('| Generated {}/{} words'.format(i, args.words))
+
+if type(model).__name__ == "TransformerSeq2Seq":
+    input = torch.randint(ntokens, (1, 1), dtype=torch.long).to(device)
+
+    with open(args.outf, 'w') as outf:
+        with torch.no_grad():  # no tracking history
+            for i in range(args.words):
+                seq_mask = model.generate_square_subsequent_mask(input.size(0)).to(device)
+                output = model(input, input, src_mask=seq_mask, tgt_mask=seq_mask)
```

Comment:
Since `seq_mask` is generated only to be fed into the model, maybe we can do this within the model forward?

Reply:
Generate seq_mask and tgt_mask is doable inside the forward function. Currently those are generated at the beginning of each epoch so they are re-used through the time steps. I thought this saves efforts.